client/header/activity-panel/panels/inbox/utils.js

```diff
@@ -15,13 +15,14 @@ export function getUnreadNotesCount( notes, lastRead ) {
 		const {
 			is_deleted: isDeleted,
 			date_created_gmt: dateCreatedGmt,
+			status,
 		} = note;
 		if ( ! isDeleted ) {
-			return (
+			const unread =
```

Comment:
One issue with doing it this way is that you could have unread or unactioned notes from 25 notes ago and this would return 0 notes because those notes aren't queried.

This may be more of an edge case, but I just want to surface this potential problem before we merge it if we think this behavior is okay.

Reply:
Hi @joshuatf thank you for the review 🙂

Since we don't have pagination (or lazyloading) implemented in the notification list, looking at all the notes to verify if any of them are `unread` could be a little bit tricky.
E.g.: If we have 27 notes to show, the only way a user is able to see the other 2 is by dismissing 2 notes. In this case, if the first notification is `unread`, the user will be notified that they have an unread message but they won't be able to see it in a normal navigation flow. They will be able to see it only by dismissing 2 notifications individually.

Having this in mind, I thought it would be better for the user to get an unread message warning for the inbox notifications that they can see/dismiss. But I really would like to know your opinion about this, taking these additional details into consideration.

I think that for a follow-up PR it would be very helpful to add pagination (or lazyloading) to the notification list. And for now, if you think that we should use all the notifications to do the check I can do it with no problem 👍

Reply:
>Since we don't have pagination (or lazyloading) implemented in the notification list

Good point.  This is not something we have to handle now, but would be nice to in the future for performance and accuracy once we have pagination of some kind.

I think a TODO comment in the indicators will be enough for now.